src/main/java/seedu/address/model/entitylist/MentorList.java

```diff
@@ -50,7 +50,7 @@ public Mentor get(Id id) throws AlfredException {
      */
     public boolean update(Id id, Mentor updatedMentor) {
```

Comment:
Here I meant that you should also check if mentorList already contains updatedMentor. For example, we might have (John, cs2103@gmail.com, 12345678) and (Joshua, cs2103@gmail.com, 12345678). If I change John's name to Joshua, then we'll essentially have two same entities.

Reply:
Hmmm but they wont have the same Id

Reply:
Its a tricky case yea

Reply:
Solved in another PR